### PR TITLE
update add-on configuration apis

### PR DESCRIPTION
**Pull request by skeeey**
Signed-off-by: Wei Liu <liuweixa@redhat.com>

### CODE REVIEW

**Comment by skeeey** [comment]
- change the AddOnDeploymentConfig to namespace scope API
- make the ClusterManagementAddOn and ManagedAddOn apis support configuration reference list

File: addon/v1alpha1/types_clustermanagementaddon.go [hunk 1]
```diff
@@ -74,14 +74,18 @@ type ConfigCoordinates struct {
 	// +optional
 	LastObservedGeneration int64 `json:"lastObservedGeneration,omitempty"`
 
-	// configGroupResource represents the GroupResource of the add-on configuration.
+	// configGroupResources is a list of configuration types supported by add-on.
+	// An empty list means the add-on does not require configurations.
+	// The default is an empty list
 	// +optional
-	ConfigGroupResource *ConfigGroupResource `json:"configGroupResource,omitempty"`
+	ConfigGroupResources []ConfigGroupResource `json:"configGroupResources,omitempty"`
```
Comment (qiujian16) [inline addon/v1alpha1/types_clustermanagementaddon.go]: seems this is duplicated with DefaultConfigs

Comment (skeeey) [reply addon/v1alpha1/types_clustermanagementaddon.go]: yes. it is, I will try to refactor this part

File: addon/v1alpha1/types_clustermanagementaddon.go [hunk 2]
```diff
@@ -37,11 +37,18 @@ type ClusterManagementAddOnSpec struct {
...
 	// In scenario where a multiple add-ons share the same add-on CRD, multiple ClusterManagementAddOn
 	// resources need to be created and reference the same AddOnConfiguration.
 	// +optional
 	AddOnConfiguration ConfigCoordinates `json:"addOnConfiguration,omitempty"`
+
+	// configs is a list of configuration types supported by add-on.
+	// An empty list means the add-on does not require configurations.
+	// The default is an empty list
+	// +optional
+	Configs []ConfigMeta `json:"configs,omitempty"`
```
Comment (qiujian16) [inline addon/v1alpha1/types_clustermanagementaddon.go]: supportedConfigs?

Comment (skeeey) [reply addon/v1alpha1/types_clustermanagementaddon.go]: chanaged

**Pull request merged**